mesonbuild/cmake/traceparser.py

```diff
@@ -138,6 +138,17 @@ def __init__(self, cmake_version: str, build_dir: Path, permissive: bool = True)
             'meson_ps_disabled_function': self._meson_ps_disabled_function,
         }  # type: T.Dict[str, T.Callable[[CMakeTraceLine], None]]
 
+        if version_compare(self.cmake_version, '<3.17.0'):
+            mlog.deprecation(textwrap.dedent(f'''\
+                CMake support for versions <3.17 is deprecated since Meson 0.62.0.
```

Comment:
Why is this moving to the traceparser from its previous location?

Reply:
Because this also affects CMake dependencies, not only CMake subprojects and the traceparser is used in both code paths.

Reply:
Ideally, this should have been in the traceparser from the start, but I didn't think that far ahead.